vault/logical_system.go

```diff
@@ -3019,28 +3019,31 @@ func (*SystemBackend) handlePoliciesPasswordSet(ctx context.Context, req *logica
 			fmt.Sprintf("passwords must be between %d and %d characters", minPasswordLength, maxPasswordLength))
 	}
 
-	// Generate some passwords to ensure that we're confident that the policy isn't impossible
-	timeout := 1 * time.Second
-	genCtx, cancel := context.WithTimeout(ctx, timeout)
-	defer cancel()
-
-	attempts := 10
-	failed := 0
-	for i := 0; i < attempts; i++ {
-		_, err = policy.Generate(genCtx, nil)
-		if err != nil {
-			failed++
+	// Attempt to construct a test password from the rules to ensure that the policy isn't impossible
+	var testPassword []rune
+	for i, rule := range policy.Rules {
+		charsetRule, ok := rule.(random.CharsetRule)
+		if !ok {
+			return nil, logical.CodedError(http.StatusBadRequest, fmt.Sprintf("unexpected rule type %T", charsetRule))
 		}
-	}
 
-	if failed == attempts {
-		return nil, logical.CodedError(http.StatusBadRequest,
-			fmt.Sprintf("unable to generate password from provided policy in %s: are the rules impossible?", timeout))
+		char := charsetRule.Chars()[0]
+		for j := 0; j < charsetRule.MinLength(); j++ {
+			testPassword = append(testPassword, char)
+		}
+
+		// fill in any remaining characters with the last rule
+		if i == len(policy.Rules)-1 && len(testPassword) < policy.Length {
+			for j := len(testPassword); j < policy.Length; j++ {
```

Comment:
Instead can we just go through the rules again (but without the min length part)?  I think what I'm getting at is that it would be fun if our algorithm here could generate realistic passwords we could actually hand out.  One last enhancement to that would be to shuffle the final result.